data/src/main/java/com/microsoft/azure/kusto/data/auth/ConnectionStringBuilder.java

```diff
@@ -97,52 +70,114 @@ private ConnectionStringBuilder() {
         this.useManagedIdentityAuth = false;
         this.useAzureCli = false;
         this.useUserPromptAuth = false;
+        this.useCertificateAuth = false;
         this.userNameForTracing = null;
         this.appendedClientVersionForTracing = null;
         this.applicationNameForTracing = null;
+        this.sendX509 = false;
+        this.initialCatalog = null;
     }
 
     private void assignValue(String rawKey, String value) {
-        rawKey = rawKey.trim().toLowerCase();
-        ConnectionStringKeyword parsedKey = stringKeywordMap.get(rawKey);
+        KnownKeywords parsedKey = KCSB_KEYWORDS.get(rawKey);
```

Comment:
Assuming `get()` throws, what's the value in the other exceptions thrown here?

Reply:
I think you wrote this comment twice